dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -606,8 +636,28 @@ HRESULT STDMETHODCALLTYPE CSearchBar::Invoke(DISPID dispIdMember, REFIID riid, L
     case DISPID_SEARCHCOMPLETE:
     case DISPID_SEARCHABORT:
         SetSearchInProgress(FALSE);
+        TrySetFocus(DISPID_SEARCHCOMPLETE);
         return S_OK;
     default:
         return E_INVALIDARG;
     }
 }
+
+void CSearchBar::TrySetFocus(UINT Source)
```

Comment:
Minor question about the naming: Did you call the function "Try..." instead of "SetFocus" (as an affirmative action), because in some cases it won't set the focus ?

I also notice that the "Source" parameter doesn't seem to be used in the function: why was it added?

Reply:
Some cases it will not set focus, for example if focus is already in the "contains text" textbox.

I want to handle OnSetFocus differently but there is currently something wrong deeper down in the Explorer tab handling and I can't resolve it here. I want to use the source later.

Reply:
If the "Source" is always one of these `DISPID_...`, I would suggest you this:
```suggestion
void CSearchBar::TrySetFocus(DISPID Source)
```
(Other code in our codebase and msdn etc. usually calls such parameter as "dispId"...)

Reply:
It's technically not a DISPID but I'm using those constants for now. Something higher up in browseui has to handle [Tab] between the sidebar and the file list.